examples/kokkos-based/matrix_vector_product_kokkos.cpp

```diff
@@ -3,33 +3,72 @@
 #include <iostream>
 
 using namespace std::experimental;
+
 int main(int argc, char* argv[]) {
   std::cout << "MatrixVectorProduct" << std::endl;
-  int N = 40;
+  int N = 10;
   int M = 20;
   Kokkos::initialize(argc,argv);
   {
     Kokkos::View<double*> x_view("X",M);
     Kokkos::View<double*> y_view("Y",N);
     Kokkos::View<float**,Kokkos::LayoutRight> A_view("A",N,M);
 
-    Kokkos::deep_copy(x_view,1.0);
-    Kokkos::deep_copy(A_view,2.0);
+    {
+      // example for y = A * x
+
+      Kokkos::deep_copy(x_view,1.0);
+      Kokkos::deep_copy(A_view,2.0);
+
+      // std::experimental::mdspan a(a_ptr,N); // Requires CDAT
+      mdspan<double, extents<dynamic_extent>> x(x_view.data(),M);
+      mdspan<double, extents<dynamic_extent>> y(y_view.data(),N);
+      mdspan<float, extents<dynamic_extent,dynamic_extent>> A(A_view.data(),N,M);
+
+      // This forwards to KokkosKernels (https://github.com/kokkos/kokkos-kernels
+      linalg::matrix_vector_product(KokkosKernelsSTD::kokkos_exec<>(),A,x,y);
+      // This forwards to KokkosKernels if LINALG_ENABLE_KOKKOS_DEFAULT is ON
+      linalg::matrix_vector_product(A,x,y);
+      linalg::matrix_vector_product(std::execution::par,A,linalg::scaled(2.0,x),y);
+      // This goes to the base implementation
+      //linalg::matrix_vector_product(std::execution::seq,A,x,y);
+
+      // note that this prints 80 for each element because of the scale(2.0, x) above
```

Comment:
Would you consider, instead of printing the elements of the vector, using `assert` or a GTest macro to test the elements of the vector?  The comments suggest that you know what the correct answer should be whenever you're printing, so why not just test instead of printing?

Reply:
Yes i can do that! I am going to address this in a separate issue.

Reply:
@mhoemmen to answer your comment I opened this: https://github.com/kokkos/stdBLAS/issues/125